test/backend_graph_comparison.in.cpp

```diff
@@ -0,0 +1,279 @@
+// Copyright 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include <cstdlib>
+#include <string>
+#include "gtest/gtest.h"
+
+#include "ngraph/log.hpp"
+#include "ngraph/ngraph.hpp"
+#include "ngraph/runtime/host_tensor.hpp"
+#include "ngraph/serializer.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/test_control.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+static string s_manifest = "${MANIFEST}";
+
+// Params to drive concat_vector_large testing variations
+class frozen_graph_files : public ::testing::TestWithParam<string>
+{
+public:
+    void compare_results(NodeVector& result_nodes,
+                         vector<shared_ptr<runtime::Tensor>> ref_results,
+                         vector<shared_ptr<runtime::Tensor>> bk_results)
+    {
+        for (int i = 0; i < ref_results.size(); ++i)
+        {
+            const shared_ptr<runtime::Tensor>& ref_data = ref_results.at(i);
+            const shared_ptr<runtime::Tensor>& bk_data = bk_results.at(i);
+
+            cout << "Comparing results for " << result_nodes.at(i)->get_name() << endl;
+            if (auto node = dynamic_pointer_cast<op::GetOutputElement>(result_nodes.at(i)))
+            {
+                cout << "  Parent node: ";
+                for (auto& p : node->get_arguments())
+                {
+                    cout << " " << p->get_name() << endl;
+                    cout << "   nargs: " << p->get_arguments().size() << endl;
+                }
+            }
+
+            ASSERT_EQ(ref_data->get_element_type(), bk_data->get_element_type());
+            ASSERT_EQ(ref_data->get_element_count(), bk_data->get_element_count());
+            ASSERT_EQ(ref_data->get_shape(), bk_data->get_shape());
+
+            element::Type et = ref_data->get_element_type();
+            if (et == element::boolean)
+            {
+                vector<char> ref_data_vector = read_vector<char>(ref_data);
+                vector<char> bk_data_vector = read_vector<char>(bk_data);
+                print_results(ref_data_vector, bk_data_vector);
+                EXPECT_TRUE(test::all_close<char>(ref_data_vector, bk_data_vector));
+            }
+            else if ((et == element::f32) || (et == element::f64))
+            {
+                vector<float> ref_data_vector = read_float_vector(ref_data);
```

Comment:
In keeping with existing implementation of all_close_f.
1. For vector params, all_close_f only defined so far with vector<float>
2. all_close_f variant which takes Tensor params reads via read_float_vector
I did read calls by hand (both here and for all_close) because the get_tensor_layout comparisons didn't make sense across runtimes.

It would make sense to extend all_close_f to test double future PR, but ideally I'd like a good test case where we can consider how tight we should be testing.